### PR TITLE
Indicates in the setting that the email is not visibile to the other users #5315

### PR DESCRIPTION
**Pull request by aka001**
The screenshot shows the changes done. Kindly review this pr. 
screenshot from 2015-02-20 03 03 05
screenshot from 2015-02-20 03 02 54

### CODE REVIEW

File: app/views/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+              <div id="terms" class="text-center">
+                <%= t('.terms', terms_link: link_to(t('.terms_link'), terms_path, target: "_blank")).html_safe %>
+              </div>
+            <% end %>
+            <%= f.submit t('.continue'), class: "new-btn", data: {disable_with: t('.submitting')} %>
+        <% end %>
+      </div>
+    </div>
+  </div>
+</div>
```
Comment (aka001) [inline app/views/registrations/new.html.erb]: Does this file need to be dropped?

Comment (jhass) [inline app/views/registrations/new.html.erb]: I guess you started working without updating from upstream first, then rebased before opening the PR? #5391 was merged today and that ported this template to HAML

Comment (aka001) [inline app/views/registrations/new.html.erb]: In that case, since there are minor changes, I will create a new pr.

Comment (svbergerem) [inline app/views/registrations/new.html.erb]: There is no need for a new PR. You can just modify the changes in this PR.

Comment (aka001) [inline app/views/registrations/new.html.erb]: I have updated the pr. Kindly look into it.
The latest changes in the registration page:
screenshot from 2015-02-20 05 41 01

**Comment by svbergerem** [comment]
Looks like you removed the wrong changes from this PR. Let's clean up this PR before talking about the code itself.
1. Remove the right file:
`git rm app/views/registrations/new.html.erb`
2. Change `app/views/users/edit.html.haml` like you did that before. (You removed those changes)
3. Revert the changes in `config/locales/diaspora/en.yml` except for the strings you added. You only need to add the string once and you can use it in both places. For example when you keep the string for `users.edit.your_email_private` you can use `t('.your_email_private')` in `app/views/users/edit.html.haml` and `t('users.edit.your_email_private')` everywhere else.

**Comment by aka001** [comment]
@svbergerem Thank you for the detailed explanation. I have updated the pr and squashed the commits. Kindly review them.

**Comment by goobertron** [comment]
I'm not sure why the heading 'Your email' appears a line below 'Your diaspora* ID'; the two should be aligned horizontally. It doesn't look as though your PR could have changed this. Could it be your browser?

**Comment by aka001** [comment]
@goobertron It was indeed my browser. In firefox the view appears as:
screenshot from 2015-02-21 03 36 50

**Comment by svbergerem** [comment]
@aka001 This is much better. For the settings page I'd prefer a lock icon with a tooltip instead of the text. That would be similar to what we do in the single post view or what @jaywink started to do in #4596. The lock icon for the single post view is added [here](https://github.com/diaspora/diaspora/blob/develop/app/assets/templates/single-post-viewer/single-post-content_tpl.jst.hbs#L35). We are using enypo icons, in this case the lock icon which you get with
```haml
%i.entypo.lock.gray
```
Just like in the single post view you could put that in a `%span` where you add the title for the tooltip.

You would have to write a small backbone view (similar to the [tags view](https://github.com/diaspora/diaspora/blob/develop/app/assets/javascripts/app/views/tags_view.js)) where you [activate the tooltip](http://getbootstrap.com/2.3.2/javascript.html#tooltips) and call it from the [router](https://github.com/diaspora/diaspora/blob/develop/app/assets/javascripts/app/router.js). That could look similar to this:


For the registration page I don't think that this is the right position for the text. I don't have a better idea but maybe @goobertron or @Flaburgan could help here. (perhaps they are even happy with the current solution)

**Comment by Flaburgan** [comment]
Yeah the place of the indication on the registration page is not good imo, because:
1. This is where we display the link to the terms
2. The user needs to have the information before entering his email, here, it's too far after.

**Comment by svbergerem** [comment]
How about a [popover](http://getbootstrap.com/2.3.2/javascript.html#popovers) on focus? (We could also use a tooltip but to me popovers looked more appropriate in this case.)


You can add the text with the `data-content` attribute to the input. The popover will automatically use the value of the title attribute (“Enter an email”) as the header of the popover. You would have to initialize the popover in a backbone view with code similar to this one:
```javascript
$('input#user_email').popover({
  placement: 'left',
  trigger: 'focus'
});
```

**Comment by Flaburgan** [comment]
yeah that's better, though I don't know what's better between popover and tooltip

**Comment by svbergerem** [comment]
This is the sign up page with a tooltip:
sign up tooltip

I'd prefer the tooltip for the settings page and the popover for the sign up page.

**Comment by goobertron** [comment]
Thanks fr supplying the ideas and screenshots, Steffen.

> I'd prefer the tooltip for the settings page and the popover for the sign up page.

I agree.

It would be nice to get some indication on the sign-up page that there *is* more information to be found, in a popover or whatever. I was wondering about using the lock icon inside the email text entry field, perhaps at the right-hand side. However, the lock icon is already used for the password fields, so that might be confusing.

**Comment by svbergerem** [comment]
@goobertron We could activate autofocus for the email input like @Flaburgan did this for the username on the login page [here](https://github.com/diaspora/diaspora/blob/develop/app/views/sessions/new.html.haml#L22). Then the popover would be visible immediately.

**Comment by goobertron** [comment]
> We could activate autofocus for the email input like @Flaburgan did this for the username on the login page here. Then the popover would be visible immediately.

Perfect!

**Comment by Flaburgan** [comment]
if the tooltip is displayed on focus, it's already fine imo.

**Comment by aka001** [comment]
Thank you everyone for your inputs. I am running a bit busy with my college projects. I will complete it by tomorrow night IST.

**Comment by svbergerem** [comment]
@aka001 What is the status? Do you need any help?

**Comment by aka001** [comment]
@svbergerem Thank you for the reply. I do need help but I will continue this work after some days and mention the details here. I am having exams in 4 days. Sorry for the delay.

**Comment by svbergerem** [comment]
Sure, no problem. Good luck!

**Comment by svbergerem** [comment]
@aka001 ping.

**Comment by aka001** [comment]
@svbergerem I have a project deadline tomorrow and will be travelling after that. I will start on this issue after 30th April. Sorry for the delay.

**Comment by svbergerem** [comment]
Superseded by #5956. Thank you for your work!

**Comment by aka001** [comment]
@svbergerem Awesome.

**Pull request closed without merging**